src/actions/contact-request/saveContactRequest.ts

```diff
@@ -1,19 +1,28 @@
 'use server'
-import { createContactRequest } from '@/backend/contact-request/contact-request.service'
+import {
+  createContactRequest,
+  findRedundantContactRequest,
+} from '@/backend/contact-request/contact-request.service'
 import { ContactFormRequest } from '@/components/ContactForm/schema'
 
 export const saveContactRequest = async (
   contactFormRequestData: ContactFormRequest,
 ) => {
   try {
-    const createdContactRequest = await createContactRequest(
-      contactFormRequestData,
-    )
-    if (createdContactRequest) {
-      createdContactRequest
-    } else {
-      return false
+    const redundantContactRequest = await findRedundantContactRequest({
+      senderEmail: contactFormRequestData.senderEmail,
+      profileId: contactFormRequestData.profileId,
+    })
+    if (!redundantContactRequest) {
+      const createdContactRequest = await createContactRequest(
+        contactFormRequestData,
+      )
+      if (createdContactRequest) {
+        return createdContactRequest
+      }
     }
+    console.log('User has already contacted this profile')
```

Comment:
**issue:** We should use toaster to notify user about it instead of logging it. We have this logic already, eg: when accepting profile in moderation zone